src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1077,7 +1077,16 @@ bool os::dll_address_to_library_name(address addr, char* buf,
     return false;
   }
 
-  return AixSymbols::get_module_name(addr, buf, buflen);
+  address  base = nullptr;
+  if ( false == AixSymbols::get_module_name_and_base(addr, buf, buflen, &base)
+       || base == nullptr ) {
+    return false;
+  }
+  if (offset != nullptr) {
+    *offset = addr - base;
```

Comment:
Maybe add a sanity check that addr >= base and <= (base + INT_MAX). If it never should happen, an assert is fine, otherwise handled at runtime.